homeassistant/components/fully_kiosk/number.py

```diff
@@ -0,0 +1,88 @@
+"""Fully Kiosk Browser number entity."""
+from __future__ import annotations
+
+from homeassistant.components.number import NumberEntity, NumberEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import TIME_SECONDS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import FullyKioskDataUpdateCoordinator
+from .entity import FullyKioskEntity
+
+ENTITY_TYPES: tuple[NumberEntityDescription, ...] = (
+    NumberEntityDescription(
+        key="timeToScreensaverV2",
+        name="Screensaver Timer",
+        native_max_value=9999,
+        native_step=1,
+        native_min_value=0,
+        native_unit_of_measurement=TIME_SECONDS,
+        entity_category=EntityCategory.CONFIG,
+    ),
+    NumberEntityDescription(
+        key="screensaverBrightness",
+        name="Screensaver Brightness",
+        native_max_value=255,
+        native_step=1,
+        native_min_value=0,
+        entity_category=EntityCategory.CONFIG,
+    ),
+    NumberEntityDescription(
+        key="timeToScreenOffV2",
+        name="Screen Off Timer",
+        native_max_value=9999,
+        native_step=1,
+        native_min_value=0,
+        native_unit_of_measurement=TIME_SECONDS,
+        entity_category=EntityCategory.CONFIG,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Fully Kiosk Browser number entities."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        FullyNumberEntity(coordinator, entity)
+        for entity in ENTITY_TYPES
+        if entity.key in coordinator.data["settings"]
+    )
+
+
+class FullyNumberEntity(FullyKioskEntity, NumberEntity):
+    """Representation of a Fully Kiosk Browser entity."""
+
+    def __init__(
+        self,
+        coordinator: FullyKioskDataUpdateCoordinator,
+        description: NumberEntityDescription,
+    ) -> None:
+        """Initialize the number entity."""
+        super().__init__(coordinator)
+        self.entity_description = description
+        self._attr_unique_id = f"{coordinator.data['deviceID']}-{description.key}"
+
+    @property
+    def native_value(self) -> int | None:
+        """Return the state of the number entity."""
+        if not self.coordinator.data:
+            return None
```

Comment:
Why is this check needed?

Reply:
forgot to remove it when i was copying from the custom component